projects/find/find.py

```diff
@@ -0,0 +1,57 @@
+import os
+import sys
+
+"""this method is created to find all file and directory, given a specific path. Optionally, your can use -name flag
+to search only the files that match to a specific name"""
+
+
+def find(path, option=None, filename=None):
+    """this method uses a bfs approach to resolve to the problem of finding all files and directories."""
+    if not os.path.exists(path):
+        return "Path non corretto o inestitente: si prega di riprovare"
+    queue = []
+    queue.append(path)
+    # marked = [False] * len(list)
+    # edge_to = [None] * len(list)
+    list = []
+    if filename is None and option is None:
+        while len(queue) > 0:
+            queue_element = queue.pop(0)
+            adj_list = os.listdir(queue_element)
+            if adj_list is not []:
+                for adj in adj_list:
+                    if os.path.isfile(queue_element + adj):
+                        # print queue_element+adj
+                        list.append(queue_element + adj)
+                    elif os.path.isdir(queue_element + adj + "/"):
+                        queue.append(queue_element + adj + "/")
+                        # print queue_element+adj
+                        list.append(queue_element + adj)
+        list.sort()
+        for path in list:
+            print path
+    elif filename is not None and option == "-name":
```

Comment:
Is it really worth to duplicate the whole algorithm just for the -name option?
I'd rather have an if when appending to the list, that way the whole thing is shorter and there's no duplicated logic anywhere